vispy/gloo/framebuffer.py

```diff
@@ -257,5 +257,4 @@ def read(self, mode='color', alpha=True, crop=None):
         
```

Comment:
We could move the `buffer = getattr(self, mode+'_buffer')` out of the `if crop` and check, if the requested buffer is available or not. If not, we would have to raise an Error I think.

Reply:
Yeah that could make sense and yes this is what I was talking about with stencil_buffer being None. I'll see what I can do.